custom/icds/models.py

```diff
@@ -0,0 +1,148 @@
+
+from casexml.apps.case.models import CommCareCase
+from corehq.apps.userreports.expressions.getters import transform_date
+
+
+from .constants import *
+
+
+class ChildHealthCaseRow(object):
+    def __init__(self, case, snapshot_start_date, snapshot_end_date, forms_provider=[]):
+        """
+        Snapshot of the case in the duration specified by snapshot_start_date and snapshot_end_date
+
+        args:
+            case: case of type 'child_health'
+            snapshot_start_date: Start date of duration to snapshot the case on
+            snapshot_end_date: End date of duration to snapshot the case on.
+
+        dates can be date/datetime objects or a valid strings
+        """
+        if not isinstance(case, CommCareCase):
+            case = CommCareCase.wrap(case)
```

Comment:
Isn't most of UCR built around the fact that the data is in json-format? What are the implications of SQL models on UCR overall? Is there doc I should be looking at, to write SQL-proof code?